aten/src/ATen/native/cpu/BlasKernel.cpp

```diff
@@ -0,0 +1,197 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/CPUBlas.h>
```

Comment:
Are you sure you actually want to vectorize these fallbacks? They're so naive I'm not sure they're worth the binary size to compile them with AVX/etc.

Reply:
Do you know if we have any test coverage for this code?

Reply:
BFloat16, Half and ints < 64 all unconditionally exercise this code path. Only BFloat16 and Half seem to actually have test coverage though.

>Are you sure you actually want to vectorize these fallbacks? They're so naive I'm not sure they're worth the binary size to compile them with AVX/etc.

There's certainly a lot of room for improvement but at the very least, I think `a.T @ b` and `b & a.T` should vectorize very well and be reasonably efficient.